actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -77,6 +77,10 @@ module RequestForgeryProtection
       config_accessor :log_warning_on_csrf_failure
       self.log_warning_on_csrf_failure = true
 
+      # Controls whether the Origin header is checked in addition to the CSRF token.
+      config_accessor :forgery_protection_origin_check
+      self.forgery_protection_origin_check = false
+
```

Comment:
Yeah, it would also break apps that might run on multiple domains and POST between them. I'll defer to you folks' judgement on whether it should be enabled by default.